internal/testutils/endpoint.go

```diff
@@ -33,21 +33,20 @@ func (b byAllFields) Less(i, j int) bool {
 		return true
 	}
 	if b[i].DNSName == b[j].DNSName {
-		if b[i].Target < b[j].Target {
-			return true
-		}
-		if b[i].Target == b[j].Target {
+		// This rather bad, we need a more complex comparison for Targets, which considers all elements
+		if b[i].Targets.Same(b[j].Targets) {
```

Comment:
How bad is this "rather bad" part? Could it lead to tests not failing when they better should?

Reply:
Probably not that bad. I had no failing tests so far, and afaik this comparison should be enough for testing. Should I removed the comment or do we want to keep it for the future until all providers have been modified for multiple targets in case we discover problems then?

Reply:
Let's keep it.